common/lib/client/realtime.js

```diff
@@ -80,24 +92,32 @@ var Realtime = (function() {
 		}
 	};
 
-	Channels.prototype.setInProgress = function(channel, inProgress) {
-		if(inProgress) {
-			this.inProgress[channel.name] = channel;
-		} else {
-			delete this.inProgress[channel.name];
-			if(Utils.isEmpty(this.inProgress)) {
-				this.emit('nopending');
-			}
+	/* Records operations currently pending on a transport; used by connectionManager to decide when
+	 * it's safe to upgrade. Note that a channel might be in the attaching state without any pending
+	 * operations (eg if attached while the connection state is connecting) - such a channel must not
+	 * hold up an upgrade, so is not considered inProgress.
+	 * Operation is currently one of either 'statechange' or 'sync' */
+	Channels.prototype.setInProgress = function(channel, operation, bool) {
+		this.inProgress[channel.name] = this.inProgress[channel.name] || {};
```

Comment:
Fair enough, I guess my reasoning for changing it away from `inProgress` was a bit flimsy. Changed back: 63be463